### PR TITLE
Add fscacher

### PR DESCRIPTION
**Pull request by jwodder**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

CC: @yarikoptic

### CODE REVIEW

File: recipes/fscacher/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  summary: Caching results of operations on heavy file trees
+  description: |
+    fscacher provides a cache & decorator for memoizing functions whose outputs
+    depend upon the contents of a file argument.
+  dev_url: https://github.com/con/fscacher
+
+extra:
+  recipe-maintainers:
+    - jwodder
+    - yarikoptic
```
Comment (yarikoptic) [inline recipes/fscacher/meta.yaml:51]: confirming willingness to participate ;)

File: recipes/fscacher/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 3a560253b5d6f8c00c1e1ff7b48731d59ce091544a13d4284c43c40b38ff6ffd
+
+build:
+  number: 0
+  skip: True  # [py<36]
```
Comment (chrisburr) [inline recipes/fscacher/meta.yaml:14]: This looks like it can be `noarch: python`:

```suggestion
  noarch: python
```

Comment (jwodder) [reply recipes/fscacher/meta.yaml:14]: No, the package requires Python 3.6 or higher.

Comment (chrisburr) [reply recipes/fscacher/meta.yaml:14]: For `noarch: python` packages you can add `- python >=3.6` in `host` and `run`. The bot will complain that it's missing as soon as this suggestion is applied.

Comment (jwodder) [reply recipes/fscacher/meta.yaml]: Done.

**Pull request merged**